launch_ros/launch_ros/actions/lifecycle_node.py

```diff
@@ -77,12 +78,35 @@ def _on_transition_event(self, context, msg):
     def _call_change_state(self, request, context: launch.LaunchContext):
         while not self.__rclpy_change_state_client.wait_for_service(timeout_sec=1.0):
             if context.is_shutdown:
-                self.___logger.warning(
+                self.__logger.warning(
                     "Abandoning wait for the '{}' service, due to shutdown.".format(
                         self.__rclpy_change_state_client.srv_name),
                 )
                 return
-        response = self.__rclpy_change_state_client.call(request)
+
+        # Asynchronously wait so that we can periodically check for shutdown.
+        event = threading.Event()
+
+        def unblock(future):
+            nonlocal event
+            event.set()
+
+        response_future = self.__rclpy_change_state_client.call_async(request)
+        response_future.add_done_callback(unblock)
```

Comment:
Do we need to use a callback? Something like `while not response_future.done()` seems simpler.

Reply:
The use of the event is so that the condition in the `while` loop below can return as soon as the call is done. We could check `.done()` but we'd have to poll it if we didn't use the event.